routing/router.go

```diff
@@ -1684,6 +1684,14 @@ type LightningPayment struct {
 	// MaxParts is the maximum number of partial payments that may be used
 	// to complete the full amount.
 	MaxParts uint32
+
+	// MaxShardAmt is the largest shard that we'll attempt to split using.
+	// If this field is set, and we need to split, rather than attempting
+	// half of the original payment amount, we'll use this value if half
+	// the payment amount is greater than it.
+	//
+	// NOTE: This field is _optional_.
+	MaxShardAmt *lnwire.MilliSatoshi
```

Comment:
nit: maybe cleaner to set to the maximum value of `lnwire.MiliSatoshi` instead of using a pointer.

Reply:
We use this same pointer-means-a-value-may-be-optional for all the other possibly optional values in `LightningPayment`.